broker/src/test/java/io/camunda/zeebe/broker/exporter/stream/ExternalExporterContainerTest.java

```diff
@@ -0,0 +1,203 @@
+    final var actualClassLoader = exporterInstance.onConfigureTCL;
+    assertThat(actualClassLoader)
+        .isSameAs(expectedClassLoader)
+        .isInstanceOf(ExporterJarClassLoader.class);
+  }
+
+  @Test
+  void shouldSetTclOnOpen(final @TempDir File jarDirectory)
+      throws ExporterLoadException, IOException {
+    // given
+    final var exporterClass = createUnloadedExporter();
+    final var jarFile = createExporterJar(new File(jarDirectory, "exporter.jar"), exporterClass);
+    final var descriptor = runtime.loadExternalExporter(jarFile, EXPORTER_CLASS_NAME);
+    final var expectedClassLoader = descriptor.newInstance().getClass().getClassLoader();
+    final var container = runtime.newContainer(descriptor);
+
+    // when
+    container.openExporter();
+
+    // then
+    final var exporterInstance = (TclExporter) container.getExporter();
+    assertThat(exporterInstance.onOpenTCL)
+        .isSameAs(expectedClassLoader)
+        .isInstanceOf(ExporterJarClassLoader.class);
+  }
+
+  @Test
+  void shouldSetTclOnExport(final @TempDir File jarDirectory)
+      throws ExporterLoadException, IOException {
+    // given
+    final var exporterClass = createUnloadedExporter();
+    final var jarFile = createExporterJar(new File(jarDirectory, "exporter.jar"), exporterClass);
+    final var descriptor = runtime.loadExternalExporter(jarFile, EXPORTER_CLASS_NAME);
+    final var expectedClassLoader = descriptor.newInstance().getClass().getClassLoader();
+    final var container = runtime.newContainer(descriptor);
+
+    // when
+    final var record = mock(TypedRecord.class);
+    // set a high position to ensure we export it
+    when(record.getPosition()).thenReturn(Long.MAX_VALUE);
+    container.exportRecord(mock(RecordMetadata.class), record);
+
+    // then
+    final var exporterInstance = (TclExporter) container.getExporter();
+    assertThat(exporterInstance.onExportTCL)
+        .isSameAs(expectedClassLoader)
+        .isInstanceOf(ExporterJarClassLoader.class);
+  }
+
+  @Test
+  void shouldSetTclOnClose(final @TempDir File jarDirectory)
+      throws ExporterLoadException, IOException {
+    // given
+    final var exporterClass = createUnloadedExporter();
+    final var jarFile = createExporterJar(new File(jarDirectory, "exporter.jar"), exporterClass);
+    final var descriptor = runtime.loadExternalExporter(jarFile, EXPORTER_CLASS_NAME);
+    final var expectedClassLoader = descriptor.newInstance().getClass().getClassLoader();
+    final var container = new ExporterContainer(descriptor);
+
+    // when
+    container.close();
+
+    // then
+    final var exporterInstance = (TclExporter) container.getExporter();
+    assertThat(exporterInstance.onCloseTCL)
+        .isSameAs(expectedClassLoader)
+        .isInstanceOf(ExporterJarClassLoader.class);
+  }
+
+  private Unloaded<TclExporter> createUnloadedExporter() {
+    return new ByteBuddy().subclass(TclExporter.class).name(EXPORTER_CLASS_NAME).make();
+  }
```

Comment:
This might need a comment describing why bytebuddy is used

Reply:
I figured the class level doc was enough - do you think it's not?

Reply:
Ah sorry, I lost that context during this review point I think., but you're absolutely right. Fine as is